internal/discharger: implement trusted domains

mhilton opened the pull request:
Add the concept of trusted domains that can be used in the redirect flow
and will allow any path with the domain, and optionally all subdomains
to be trusted return_to addresses.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/107/

urosj commented on internal/v1/discharge.go:
```diff
@@ -120,7 +120,13 @@ func checkThirdPartyCaveat(ctx context.Context, h *handler, req *http.Request, c
 	switch cond {
-	case "is-authenticated-user":
+	case "is-authenticated-user", "!is-authenticated-user":
+		// Note above that we recognize but ignore a "!" prefix
```
Could we use '<' instead, saying it's "smaller" version of the the protocol?

rogpeppe commented:
this was merged into the blues-identity repo instead.

mhilton committed:
internal/discharger: implement trusted domains

Add the concept of trusted domains that can be used in the redirect flow
and will allow any path with the domain, and optionally all subdomains
to be trusted return_to addresses.
